agent/configmgr/git.go

```diff
@@ -192,30 +192,39 @@ func (gc *gitConfigManager) processSelector(file *object.File, cfg config.Config
 		return nil, err
 	}
 
-	// Check for matching selector
+	// Use a set (map) to store unique policy paths
+	policyPathsSet := make(map[string]struct{})
+
+	// Iterate through all selectors and collect all matching ones
 	for selectorName, entry := range selectors {
 		matches := true
-		for key, value := range entry.Selector {
-			if cfgValue, exists := cfg.OrbAgent.Labels[key]; !exists || cfgValue != value {
-				matches = false
-				break
+		// If selector is empty, it matches everything
+		if len(entry.Selector) > 0 {
+			for key, value := range entry.Selector {
+				if cfgValue, exists := cfg.OrbAgent.Labels[key]; !exists || cfgValue != value {
+					matches = false
+					break
+				}
 			}
 		}
-
 		if matches {
 			gc.logger.Info("Selector matched", zap.String("selector", selectorName))
-			policyPaths := make([]string, 0)
 			for _, policy := range entry.Policies {
 				if policy.Enabled != nil && !*policy.Enabled {
 					continue
 				}
-				policyPaths = append(policyPaths, policy.Path)
+				policyPathsSet[policy.Path] = struct{}{}
```

Comment:
It looks possible to have a selector with two policies pointing to the same path. Shouldn't we a least log a warning (or fail fast) in such scenario?

Reply:
added warn message! Thanks